File: src/Java/Geovis/EarthSource.java
import vtk.vtkActor;
import vtk.vtkEarthSource;
import vtk.vtkNativeLibrary;
import vtk.vtkPolyDataMapper;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkNamedColors;

 
public class EarthSource
{	 
  //-----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------
  

  public static void main(String s[]) 
  {
    double MyColors[] = new double[4];
    vtkNamedColors Colors = new vtkNamedColors();
    Colors.GetColor("Banana",MyColors);

    //Renderer Background Color
    double Bgcolor[] = new double[4];
    Colors.GetColor("Black", Bgcolor);

    vtkEarthSource EarthSource = new vtkEarthSource();
    EarthSource.OutlineOff();
    //EarthSource.OutlineOn();
    EarthSource.Update();

    //Create a Mapper and Actor
    vtkPolyDataMapper Mapper = new vtkPolyDataMapper();
    Mapper.SetInputConnection(EarthSource.GetOutputPort());

    vtkActor Actor = new vtkActor();
    Actor.SetMapper(Mapper);
    Actor.GetProperty().SetColor(MyColors);

    //Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);

    // Visualise the arrow
    ren.AddActor(Actor);
    ren.SetBackground(Bgcolor);

    renWin.SetSize(300, 300);
    renWin.Render();

    iren.Initialize();
    iren.Start();

  }
}

File: src/Python/Geovis/EarthSource.py
#!/usr/bin/env python
import vtk

def main():
    colors = vtk.vtkNamedColors()
    #Create a sphere
    earthSource = vtk.vtkEarthSource()
    earthSource.OutlineOff()
    earthSource.Update()
    
    #Create a mapper and actor
    mapper = vtk.vtkPolyDataMapper()
    mapper.SetInputConnection(earthSource.GetOutputPort())
    
    actor = vtk.vtkActor()
    actor.SetMapper(mapper)
    
    #Create a renderer, render window, and interactor
    renderer = vtk.vtkRenderer()
    renderWindow = vtk.vtkRenderWindow()
    renderWindow.AddRenderer(renderer)
    renderWindowInteractor = vtk.vtkRenderWindowInteractor()
    renderWindowInteractor.SetRenderWindow(renderWindow)
    
    #Add the actor to the scene
    renderer.AddActor(actor)
    renderer.SetBackground(colors.GetColor3d("green")) # Background color green
    
    #Render and interact
    renderWindow.Render()
	# screenshot code:
    w2if = vtk.vtkWindowToImageFilter()
    w2if.SetInput(renderWindow)
    w2if.Update()
    
    writer = vtk.vtkPNGWriter()
    writer.SetFileName("TestEarthSource.png")
    writer.SetInputConnection(w2if.GetOutputPort())
    writer.Write()
	
	# begin interaction
    renderWindowInteractor.Start()

if __name__ == '__main__':
    main()
